lib/prometheus/middleware/exporter.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: UTF-8
+
+require 'prometheus/client'
+require 'prometheus/client/formats/text'
+
+module Prometheus
+  module Middleware
+    # Exporter is a Rack middleware that provides a sample implementation of a
+    # Prometheus HTTP exposition endpoint.
+    #
+    # By default it will export the state of the global registry and expose it
+    # under `/metrics`. Use the `:registry` and `:path` options to change the
+    # defaults.
+    class Exporter
+      attr_reader :app, :registry, :path
+
+      FORMATS  = [Client::Formats::Text].freeze
+      FALLBACK = Client::Formats::Text
+
+      def initialize(app, options = {})
+        @app = app
+        @registry = options[:registry] || Client.registry
+        @path = options[:path] || '/metrics'
+        @acceptable = build_dictionary(FORMATS, FALLBACK)
+      end
+
+      def call(env)
+        if env['PATH_INFO'] == @path
+          format = negotiate(env['HTTP_ACCEPT'], @acceptable)
```

Comment:
I think it will be more efficient to fetch a key from `env`, like `env.fetch('HTTP_ACCEPT', '')` and assume that in `negotiate` method we will always get a string in `accept` argument.

Then we can improve `negotiate` method and replace `accept.to_s.empty?` on `accept.empty?`